### PR TITLE
[Manta] Event trigger verification in benchmarks! macro for pallet-manta-pay

### LINKED ISSUE #34: [Manta] Pallet-Manta-Pay event trigger verifications for benchmarks.

### PR DESCRIPTION
**Pull request by ghzlatarev**
closes #34 :
* Adding event trigger verification in the benchmarks! macro for pallet-manta-pay, as per substrate pallet examples.

### CODE REVIEW

**Comment by ghzlatarev** [comment]
To be frank, i'm realizing now that these checks might be redundant next to the other things we verify, but still do not hurt?

**Review by zhenfeizhang — APPROVED** [review]
LGTM. Thanks for spotting this.



> To be frank, i'm realizing now that these checks might be redundant next to the other things we verify, but still do not hurt?

In fact, we should have checked more things (see the w3f-deliverable branch)
I turned those off just to get an end-to-end benchmark running.
feel free to bring those checks back.

File: src/benchmark.rs [hunk 1]
```diff
@@ -55,16 +55,23 @@ fn benchmark_helper<T: Config>(sender: T::Origin) {
...
 
 	init_asset {
 		let caller: T::AccountId = whitelisted_caller();
-	}: init_asset (RawOrigin::Signed(caller.clone()), 1000u64)
+		let total = 1000u64;
+	}: init_asset (RawOrigin::Signed(caller.clone()), total)
 	verify {
-		assert_eq!(
-			//<TotalSupply>::get(), 1000u64
-			1000u64, 1000u64
```
Comment (ghzlatarev) [inline src/benchmark.rs]: This assertion is not failing, but let me know if I should revert it.

**Comment by zhenfeizhang** [comment]
Please run the formater again...

File: src/benchmark.rs (hunk 1 above)
Comment (zhenfeizhang) [reply src/benchmark.rs]: Yes I believe so. 
should indeed check that total supply is 1000.

**Comment by ghzlatarev** [comment]
> Please run the formater again...

I wonder why the check is passing if it is not formatted properly.

**Review by stechu — APPROVED** [review]
LGTM. I tested locally, the benchmark works.

**Pull request merged**